docs/src/main/tut/schema-evolution/avro.md

```diff
@@ -0,0 +1,292 @@
+---
+layout: docs
+title: Schema Evolution - Avro
+permalink: /schema-evolution/avro
+---
+
+# Avro - Schema Evolution
+
+From now on, consider that we are using `AvroWithSchema` as the serialization mechanism in your [mu] program.
+
+According to the [Avro Specs](http://avro.apache.org/docs/current/spec.html#Schema+Resolution):
+
+> A reader of Avro data, whether from an RPC or a file, can always parse that data because its schema is provided. But that schema may not be exactly the schema that was expected. For example, if the data was written with a different version of the software than it is read, then records may have had fields added or removed.
+
+For Scala, this section specifies how such schema differences should be resolved to preserve compatibility. We'll try to summarise a bit all the possible cases in both ends: request and response. However, you could go deeper by using this [repo](https://github.com/higherkindness/mu-protocol-decimal-update) where you can play with all of the possibilities.
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+## Cases
+
+- [Modifying the Request (Client side)](#modifying-the-request-client-side)
+  - [A: Adding a new non-optional field](#a-adding-a-new-non-optional-field)
+  - [B: Adding a new optional field](#b-adding-a-new-optional-field)
+  - [C: Adding new item to a coproduct](#c-adding-new-item-to-a-coproduct)
+  - [D: Removing item from a coproduct](#d-removing-item-from-a-coproduct)
+  - [E: Replacing item in a coproduct](#e-replacing-item-in-a-coproduct)
```

Comment:
I would say `union` instead of `coproduct` because we are talking in terms of the `avro` protocol

Reply:
Done in https://github.com/higherkindness/mu/pull/481/commits/4a79efa8d256e4cb44551a5b2f8c1aa5caa55c24